packages/compiler/src/core/helpers/type-name-utils.ts

```diff
@@ -177,12 +177,25 @@ function getInterfaceName(iface: Interface, options: TypeNameOptions | undefined
     interfaceName += `<${iface.templateMapper.args
       .map((x) => getTypeName(x, options))
       .join(", ")}>`;
+  } else if (iface.node.templateParameters.length > 0) {
+    // template
+    const params = iface.node.templateParameters.map((t) => getIdentifierName(t.id.sv, options));
+    interfaceName += `<${params.join(", ")}>`;
   }
   return `${getNamespacePrefix(iface.namespace, options)}${interfaceName}`;
 }
 
 function getOperationName(op: Operation, options: TypeNameOptions | undefined) {
-  return `${getNamespacePrefix(op.namespace, options)}${getIdentifierName(op.name, options)}`;
+  let opName = getIdentifierName(op.name, options);
+  if (op.node.templateParameters.length > 0) {
```

Comment:
@timotheeguerin , I found the operation's template instance would be true if it's parent interface is templated and referenced by an alias, and I can't find other cases that the operation's template instance would be true, so I didn't check the isTemplateInstance here. Please let me know if my understanding is not complete and more handling are needed. thanks.